handler/incoming_instance_handler.go

```diff
@@ -111,30 +105,31 @@ func (hdlr *InstanceEventHandler) validate(newInstance event.NewInstance) error
 	return nil
 }
 
-func (hdlr *InstanceEventHandler) insertDimensions(instance *model.Instance, dimensions []*model.Dimension) error {
+// func (hdlr *InstanceEventHandler) insertDimensions(ctx context.Context, instance *model.Instance, dimensions []*model.Dimension) error {
+func (hdlr *InstanceEventHandler) insertDimensions(ctx context.Context, instance *model.Instance, dimensions []*model.Dimension) error {
 
 	cache := make(map[string]string)
 
 	for _, dimension := range dimensions {
 
-		dimension, err := hdlr.Store.InsertDimension(context.Background(), cache, instance, dimension)
+		dbDimension, err := hdlr.Store.InsertDimension(context.Background(), cache, instance.DbModel().InstanceID, dimension.DbModel())
```

Comment:
Wary of creating new context values when we should be re-using the current context passed into function